src/index.js

```diff
@@ -1,8 +1,13 @@
 /* (c) 2015 Ari Porad (@ariporad) <http://ariporad.com>. License: ariporad.mit-license.org */
-import Module from 'module';
+import BuiltinModule from 'module';
 import path from 'path';
 import nodeModulesRegex from 'node-modules-regexp';
 
+// Guard against poorly mocked module constructors.
+const Module = module.constructor.length > 1
+  ? module.constructor
+  : BuiltinModule;
+
```

Comment:
The `module.constructor.length` guards against Jest which mocks `module` as just a plain object literal `{}`.